src/Symfony/Bundle/FrameworkBundle/Command/CachePoolInvalidateTagsCommand.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Command;
+
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Contracts\Cache\TagAwareCacheInterface;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ */
+#[AsCommand(name: 'cache:pool:invalidate-tags', description: 'Invalidate cache tags for all or a specific pool')]
+final class CachePoolInvalidateTagsCommand extends Command
+{
+    private iterable $pools;
+
+    /**
+     * @param iterable<string, TagAwareCacheInterface> $pools
+     */
+    public function __construct(iterable $pools)
+    {
+        parent::__construct();
+
+        $this->pools = $pools;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure(): void
+    {
+        $this
+            ->addArgument('tags', InputArgument::IS_ARRAY | InputArgument::REQUIRED, 'The tags to invalidate')
+            ->addOption('pool', 'p', InputOption::VALUE_REQUIRED, 'The pool to invalidate on')
+            ->setHelp(<<<'EOF'
+                The <info>%command.name%</info> command invalidates tags from taggable pools. By default, all pools
+                have the passed tags invalidated. Pass <info>--pool=my_pool</info> to invalidate tags on a specific pool.
+
+                  php %command.full_name% tag1 tag2
+                  php %command.full_name% tag1 tag2 --pool=my_pool
+                EOF)
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output);
+        $tags = $input->getArgument('tags');
+        $tagList = implode(', ', $tags);
+
+        foreach ($this->poolsToInvalidate($input->getOption('pool')) as $name => $pool) {
+            $io->comment(sprintf('Invalidating tag(s): <info>%s</info> from pool <comment>%s</comment>.', $tagList, $name));
+
+            if (!$pool->invalidateTags($tags)) {
+                throw new \Exception(sprintf('Cache tag(s) "%s" could not be invalidated for pool "%s".', $tagList, $name));
+            }
+        }
+
+        $io->success('Successfully invalidated cache tags.');
+
+        return 0;
+    }
+
+    /**
+     * @return iterable<string, TagAwareCacheInterface>
+     */
+    private function poolsToInvalidate(?string $name): iterable
+    {
+        if (!$name) {
+            return $this->pools;
+        }
+
+        foreach ($this->pools as $key => $pool) {
```

Comment:
instead of iterating over all pools here, what about injection a ServiceProviderInterface to the constructor?

Reply:
I made this change. What do you think about dropping the pass entirely and just using `tagged_locator('cache.taggable', 'pool')`? (We would lose the ability to ensure the pools are `TagAwareCacheInterface`'s at compile time)

Reply:
good idea, we should not ensure pools are TagAwareCacheInterface, that's not what DI config is for